src/model_lifecycle.cc

```diff
@@ -650,9 +646,12 @@ ModelLifeCycle::UpdateModelConfig(
     return;
   }
 
-  // Update model instance group.
-  Status status =
-      model->UpdateInstanceGroup(new_model_config, &model_info_lock);
+  // Update model instance group. It is the responsibility of the model to
```

Comment:
You should describe why is it okay to release `model_info->mtx_` when calling "UpdateInstanceGroup" with this function's perspective. Why was it held in the first place and why was it released...

Reply:
Yes. Updated: [Remove lock returned from scheduler update](https://github.com/triton-inference-server/core/pull/207/commits/092a2581d221f8c78d1019a092682a1ada29adce)